### PR TITLE
[7.x] Support PostgreSQL operator containing a question mark using PHP 7.4

### PR DESCRIPTION
**Pull request by cmizzi**
Since the arrival of PHP 7.4, it is now possible to use special PostgreSQL operators such as `?`, `?|` or `?&`. This PR, therefore, includes the use of these operators by doubling the question mark.

```php
\App\User::where('roles', '?', 'admin')->get();

// ->toSql()       select * from "users" where "roles" ?? ?
// ->getBindings() ['admin']
```

For PHP < 7.4, operators don't work anyway, the use of operator functions is mandatory.

---

- https://wiki.php.net/rfc/pdo_escape_placeholders ;

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Tests failing.

**Comment by cmizzi** [comment]
> Tests failing.

Fixed.

File: src/Illuminate/Database/Query/Builder.php [hunk 1]
```diff
@@ -753,6 +753,10 @@ public function prepareValueAndOperator($value, $operator, $useDefault = false)
             throw new InvalidArgumentException('Illegal operator and value combination.');
         }
 
+        if (strpos($operator, '?') !== false) {
+            $operator = str_replace('?', '??', $operator);
+        }
+
```
Comment (mfn) [inline src/Illuminate/Database/Query/Builder.php:759]: Uh ... 
image

Comment (cmizzi) [reply src/Illuminate/Database/Query/Builder.php:759]: Why ? The RFC stipulates that anything containing a single question mark on the operator must be duplicated in order not to be considered as a binding. Are there cases where the operator cannot be a string?

**Comment by taylorotwell** [comment]
Send to master branch please.

File: src/Illuminate/Database/Query/Builder.php (hunk 1 above)
Comment (tben) [reply src/Illuminate/Database/Query/Builder.php:759]: The assumption is that the user might not want to escape the question mark. But what happens if they have? Would the suggestion below be better? 

if ($operator === "?") {
    $operator = str_replace('?', '??', $operator);
}

**Pull request closed without merging**